src/prefect/cli/deployment.py

```diff
@@ -258,60 +258,58 @@ def _load_deployments(path: Path, quietly=False) -> PrefectObjectRegistry:
 
 @deployment_app.command()
 async def apply(
-    path: Path = typer.Argument(
-        None,
-        help="The path to a deployment YAML file.",
-        show_default=False,
-    )
+    paths: List[str] = typer.Argument(
+        ...,
+        help="One or more paths to deployment YAML files.",
+    ),
 ):
     """
     Create or update a deployment from a YAML file.
     """
-    if path is None:
-        path = "deployment.yaml"
+    for path in paths:
 
-    # load the file
-    with open(str(path), "r") as f:
-        data = yaml.safe_load(f)
+        # load the file
+        with open(str(path), "r") as f:
+            data = yaml.safe_load(f)
 
-    # create deployment object
-    try:
-        deployment = DeploymentYAML(**data)
-        app.console.print(f"Successfully loaded {deployment.name!r}", style="green")
-    except Exception as exc:
-        exit_with_error(f"Provided file did not conform to deployment spec: {exc!r}")
-
-    async with get_client() as client:
-        # prep IDs
-        flow_id = await client.create_flow_from_name(deployment.flow_name)
+        # create deployment object
+        try:
+            deployment = DeploymentYAML(**data)
+            app.console.print(f"Successfully loaded {deployment.name!r}", style="green")
+        except Exception as exc:
+            exit_with_error(f"'{path!s}' did not conform to deployment spec: {exc!r}")
+
+        async with get_client() as client:
+            # prep IDs
+            flow_id = await client.create_flow_from_name(deployment.flow_name)
+
+            deployment.infrastructure = deployment.infrastructure.copy()
```

Comment:
can someone link deployment infrastructure to a stable block?
im confused here
(no change in logic im just curious)

Reply:
I had planned for that to be the case eventually, but needed to set the run command and image tag (for container based runs) at deployment creation time which required a copy. If we have the manifest attached to the deployment now separately, we could probably load those fields from there at runtime instead and just override whatever is on the block to support a stable block. It gets a little confusing though.